client/jetpack-connect/plans.jsx

```diff
@@ -230,19 +228,28 @@ class Plans extends Component {
 		mc.bumpStat( 'calypso_jpc_plan_selection', cartItem.product_slug );
 
 		addItem( cartItem );
-		this.redirecting = true;
 		this.props.completeFlow();
 		page.redirect( checkoutPath );
 	};
 
 	render() {
-		const { interval, isRtlLayout, selectedSite, translate } = this.props;
+		const {
+			canPurchasePlans,
+			hasPlan,
+			interval,
+			isAutomatedTransfer,
+			isRtlLayout,
+			notJetpack,
+			selectedSite,
+			translate,
+		} = this.props;
 
 		if (
-			this.redirecting ||
-			this.hasPreSelectedPlan() ||
-			! this.props.canPurchasePlans ||
-			this.props.hasPlan
+			this.hasPreSelectedPlan( this.props ) ||
```

Comment:
Could we get into a state where we know `hasPreSelectedPlan === true`, but `componentWillReceiveProps` won't trigger `maybeRedirect` because none of the compared props are checked?

Reply:
I don't want to add anything else to this PR, but we might make `hasPreSelectedPlan` a connected prop in a followup.

Reply:
Ff either `flowType` or `selectedPlan` change, `maybeRedirect()` will be called. Making `hasPreSelectedPlan` a connected prop would make this clearer, I think we should do it in this PR.

Reply:
Mmm, on second thoughts, I'm reluctant to change this PR further. I have a commit, I'll save it for another PR.

Reply:
Yeah, let's leave it for follow up. This is in a good place and I've tested. We can 🚢  and iterate.